services/alert/service.go

```diff
@@ -1017,10 +1017,20 @@ func (s *Service) createHandlerFromSpec(spec HandlerSpec) (handler, error) {
 	default:
 		err = fmt.Errorf("unsupported action kind %q", spec.Kind)
 	}
+	if h == nil && err != nil {
+		return handler{}, err
+	}
 	if spec.Match != "" {
 		// Wrap handler in match handler
 		handlerDiag := s.diag.WithHandlerContext(ctx...)
-		h, err = newMatchHandler(spec.Match, h, handlerDiag)
+		if h == nil {
+			panic("handler is nil, this should not happen")
```

Comment:
Does this one need to be a panic? Could we return an `err` instead?

Reply:
It is a state that should never happen and should panic.  The problem was it was panic-ing in a very distant place and made it hard to debug.